src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -939,12 +964,19 @@ private void ReduceCountBeforeAppend()
                 _tss.Stats.UpdateCountOfExistingStats(_context, SliceHolder, _collection, -ReadOnlySegment.NumberOfLiveEntries);
             }
 
-            public long AppendExistingSegment(TimeSeriesValuesSegment newValueSegment)
+            public long AppendExistingSegment(TimeSeriesValuesSegment newValueSegment, bool fromUpdateStatus = false, string changeVectorFromReplication = null)
             {
                 if (newValueSegment.RecomputeRequired)
                     newValueSegment = newValueSegment.Recompute(_context.Allocator);
 
-                (_currentChangeVector, _currentEtag) = _tss.GenerateChangeVector(_context);
+                if (fromUpdateStatus)
+                {
+                    _currentEtag = _tss._documentsStorage.GenerateNextEtag();
+                    changeVectorFromReplication ??= _tss._documentsStorage.GetNewChangeVector(_context, _currentEtag);
```

Comment:
why do I set here `changeVectorFromReplication`? I should only get it external, no?

Reply:
and why do I need it?

Reply:
in case of an update - we should use the `changeVectorFromReplication` instead of generating a new one - because we accept the changes that already exists on other nodes - so we don't need to inform others about our changes back (in that point, there's no need in more replication ping-pong - we can stop here)

Reply:
Do we really need `conflictStatusIsUpdate`? can't we simply rely on `changeVectorFromReplication` ? 
if it isn't `null` use it, otherwise generate new one.

Reply:
done